src/main/java/org/cryptomator/windows/common/RegistryKey.java

```diff
@@ -0,0 +1,171 @@
+				MemorySegment.copy(lpData, 0L, returnBuffer, 0L, returnBuffer.byteSize());
+				return returnBuffer;
+			} catch (BufferTooSmallException _) {
+				if (bufferSize <= WindowsRegistry.MAX_DATA_SIZE) {
+					return getValue(lpValueName, dwFlags, seed << 1);
+				} else {
+					throw new RuntimeException("Getting value %s for key %s failed. Maximum buffer size of %d reached.".formatted(lpValueName.getString(0, StandardCharsets.UTF_16LE), path, bufferSize));
+				}
+			}
+		}
+	}
+
+	private static class BufferTooSmallException extends RuntimeException {
+
+	}
+
+	//-- SetValue functions --
+
+	public void setStringValue(String name, String data, boolean isExpandable) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var lpValueName = arena.allocateFrom(name, StandardCharsets.UTF_16LE);
+			var lpValueData = arena.allocateFrom(data, StandardCharsets.UTF_16LE);
+			setValue(lpValueName, lpValueData, isExpandable ? REG_EXPAND_SZ() : Winreg_h.REG_SZ());
+		}
+	}
+
+	public void setDwordValue(String name, int data) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var lpValueName = arena.allocateFrom(name, StandardCharsets.UTF_16LE);
+			var lpValueData = arena.allocateFrom(ValueLayout.JAVA_INT, data);
+			setValue(lpValueName, lpValueData, Winreg_h.REG_DWORD());
+		}
+	}
+
+	private void setValue(MemorySegment lpValueName, MemorySegment data, int dwFlags) throws RegistryValueException {
+		if (data.byteSize() > WindowsRegistry.MAX_DATA_SIZE) {
+			throw new IllegalArgumentException("Data must be smaller than " + WindowsRegistry.MAX_DATA_SIZE + "bytes.");
+		}
+
+		int result = Winreg_h.RegSetKeyValueW(handle, NULL, lpValueName, dwFlags, data, (int) data.byteSize());
+		if (result != ERROR_SUCCESS()) {
+			throw new RegistryValueException("winreg_h:RegSetKeyValueW", path, lpValueName.getString(0, StandardCharsets.UTF_16LE), result);
+		}
+	}
+
+	//-- delete operations
+
+	public void deleteValue(String valueName) throws RegistryValueException {
+		deleteValue(valueName, false);
+	}
+
+	public void deleteValue(String valueName, boolean ignoreNotExisting) throws RegistryValueException {
+		try (var arena = Arena.ofConfined()) {
+			var lpValueName = arena.allocateFrom(valueName, StandardCharsets.UTF_16LE);
+			int result = Winreg_h.RegDeleteKeyValueW(handle, NULL, lpValueName);
+			if (result != ERROR_SUCCESS() //
+					&& !(result == ERROR_FILE_NOT_FOUND() && ignoreNotExisting)) {
+				throw new RegistryValueException("winreg_h:RegSetKeyValueW", path, lpValueName.getString(0, StandardCharsets.UTF_16LE), result);
+			}
+		}
+	}
+
+	public void deleteSubtree(String subkey) throws RegistryKeyException {
+		if (subkey == null || subkey.isBlank()) {
+			throw new IllegalArgumentException("Subkey must not be empty");
+		}
+		deleteValuesAndSubtrees(subkey);
+	}
+
+	public void deleteAllValuesAndSubtrees() throws RegistryKeyException {
+		deleteValuesAndSubtrees("");
+	}
```

Comment:
I would remove this method, as it is very prone to accidentally delete the wrong data. Better always force the caller to explicitly mention what it tries to delete.

Reply:
I keep the method. Not only saves us code in the `ExplorerQuickAccessService`, but also increases resiliency. A registry key can only be removed, if it has no subkeys.  If our quick access entry is extended by a foreign key, we would be unable to remove it again by just "reversing" our actions. A subkey listing would be required, adding more code and complexity.

Reply:
In a session with @overheadhunter we simplified the method (and lowered the maximum data size to set/get for reg values). Commited in b2856be9ce5d4b2a3ba108ca084eab15ca4ab36c